### PR TITLE
RFC: add 'QuickFixJump' autocmd event

### PR DESCRIPTION
**Pull request by martinparadiso**
# RFC: add `'QuickFixJump'` autocmd event

## Context and motivation

I was trying to implement a feature similar to LazyGit's file
view: populate the quickfix list with the output of `git status`
and then every time a new element is selected, trigger `fugitive`'s
`Gvdiffsplit` (or some other diff utility).

Like this (which I ended implemented by overriding keybindings):

[asciicast](https://asciinema.org/a/Zkh3U8GPJufEGULjQfUXqZMUV)

Turns out there is no Event for a new quickfix element being selected,
so I started digging and comeup with this draft.

## Use cases

I came up with the following possible uses:

- The scenario described above.
- If the quickfix is loaded with long/multiline compile errors, 
  every time the user jumps to a new element, a float is open
  on top of the cursor  with the full message.
- `Neotest` currently does something similar to the previous example,
  when a test fails, it goes to the failed test and opens a popup, 
  but only when the error is accessed through the `Neotest` summary window.

To summarize the main use would be opening extended information
about the currently selected qf. element.

## Comments

It's the first time I touch the nvim codebase, so I just
threw the `apply_autocmd` after the quickfix successfully
jumps to an element.

Also, since this a RFC I haven't implemented any tests nor
documentation; if there is interest in the feature, I'll 
get on it.

### CODE REVIEW

**Comment by justinmk** [comment]
`getqflist({'idx':0})` returns the currently selected quickfix item. Why can't that be used?

**Comment by martinparadiso** [comment]
> `getqflist({'idx':0})` returns the currently selected quickfix item. Why can't that be used?

As far as I can tell, that only returns the current element,
but there is no way to make an `autocmd` around it; nor detect the user
intentions. My goal is to to trigger it if the user *jumps* to an entry.
For instance `:cc`keeps the same `idx`, but the user wants 
to re-focus on the current entry.

Once in the event callback you have to effectively call
`getqflist({'idx':0})` to get the current item.

---

Just to add volume, here is another person asking for a feature
like this (3 years old):

https://www.reddit.com/r/neovim/comments/tzr8tm/autocommand_for_quickfixlist/

**Comment by justinmk** [comment]
A BufEnter handler seems to work reasonably well:

    autocmd BufEnter * echom localtime() getqflist({'idx':0})

Of course a purpose-built event is sometimes more reliable, but the use case here is not compelling enough to deviate from Vim. Send this to https://github.com/vim/vim/
